lms/djangoapps/verify_student/views.py

```diff
@@ -515,7 +525,8 @@ def _display_steps(self, always_show_payment, already_verified, already_paid):
             # The "make payment" step doubles as an intro step,
             # so if we're showing the payment step, hide the intro step.
             remove_steps |= set([self.INTRO_STEP])
-
+        if CourseMode.mode_for_course(course_key, CourseMode.NO_ID_PROFESSIONAL_MODES[0]):
```

Comment:
I would replace this check with something like:
```
if not CourseMode.has_verified_mode(...):
```

That way, this code will continue to work correctly if we add additional non-verified modes in the future.